File: android/app/src/main/kotlin/com/kenneth/stockcalc/domain/model/Trade.kt
package com.kenneth.stockcalc.domain.model

import kotlinx.datetime.Instant

enum class TradeStatus { OPEN, CLOSED }

data class StopLossEntry(
    val price: Double,
    val date: Instant,
    val note: String,
)

data class Trade(
    val id: String,
    val symbol: String,
    val entryPrice: Double,
    val shares: Int,
    val initialStopLoss: Double,
    val currentStopLoss: Double,
    val targetPrice: Double?,
    val status: TradeStatus,
    val riskAmount: Double,
    val createdAt: Instant,
    val stopLossHistory: List<StopLossEntry>,
    val exitPrice: Double? = null,
    val pnl: Double? = null,
    val closedAt: Instant? = null,
) {
    val isRiskFree: Boolean get() = currentStopLoss >= entryPrice
    val nativeCurrency: Currency get() = Currency.fromSymbol(symbol)
}

File: src/types/trade.ts
export type TradeStatus = 'open' | 'closed';
export type Currency = 'USD' | 'HKD';

export interface StopLossEntry {
  price: number;
  date: string;
  note: string;
}

export interface Trade {
  id: string | number;
  symbol: string;
  entryPrice: number;
  shares: number;
  initialStopLoss: number;
  currentStopLoss: number;
  targetPrice: number | null;
  status: TradeStatus;
  riskAmount: number;
  createdAt: string;
  stopLossHistory: StopLossEntry[];
  exitPrice?: number | null;
  pnl?: number | null;
  closedAt?: string | null;
}

export interface PriceData {
  price: number;
  change: number;
  changePercent: number;
  updatedAt: string;
}
